api/v1alpha1/promotion_types.go

```diff
@@ -138,10 +138,38 @@ type PromotionStatus struct {
 	// by this Promotion as well as any additional Freight that is carried over
 	// from the target Stage's current state.
 	FreightCollection *FreightCollection `json:"freightCollection,omitempty" protobuf:"bytes,7,opt,name=freightCollection"`
+	// HealthChecks contains the health check directives to be executed after
+	// the Promotion has completed.
+	HealthChecks []HealthCheckStep `json:"healthChecks,omitempty" protobuf:"bytes,8,rep,name=healthChecks"`
 	// FinishedAt is the time when the promotion was completed.
 	FinishedAt *metav1.Time `json:"finishedAt,omitempty" protobuf:"bytes,6,opt,name=finishedAt"`
 }
 
+// HealthCheckStep describes a health check directive which can be executed by
+// a Stage to verify the health of a Promotion result.
+type HealthCheckStep struct {
```

Comment:
I think we could probably make the engine use this type directly instead of `directive.HealthCheckStep`. The same can probably be said for promotion steps.

I can handle that _after_ this PR is merged since I am still actively working on the directives package and want to avoid creating merge conflicts over there.